File: core/src/main/java/com/billiamram/framework/Presenter.java
package com.billiamram.framework;

public abstract class Presenter<T extends View> {
    protected T mView;

    public void loadView(T view) {
        mView = view;
        onViewLoaded(view);
    }

    protected abstract void onViewLoaded(T view);

    public void unloadView() {
        mView = null;
        onViewUnloaded();
    }

    protected abstract void onViewUnloaded();
}

File: core/src/main/java/com/billiamram/framework/Presenter.kt
package com.billiamram.framework

abstract class Presenter<T : View> {
    protected var mView: T? = null

    fun loadView(view: T) {
        mView = view
        onViewLoaded(view)
    }

    protected abstract fun onViewLoaded(view: T)

    fun unloadView() {
        mView = null
        onViewUnloaded()
    }

    protected abstract fun onViewUnloaded()
}
